agenda.php

```diff
@@ -241,18 +241,45 @@ protected function updateEvent(array $event) {
                 }
             }
         }
-
+        
         $this->pdo->beginTransaction();
+        
+        $query = $this->pdo->prepare("SELECT datetime_begin FROM events WHERE vCalendarFilename=:vCalendarFilename;");
+        $query->execute(array(
+            'vCalendarFilename' =>  $event['vCalendarFilename']
+        ));
+        $results = $query->fetch();
+        
+        $previous_datetime = null;
+        $new_event = false;
+        if($results === false) {
+            $new_event = true;
+        } else {
+            $previous_datetime = DateTimeImmutable::createFromFormat("Y-m-d H:i:s", $results["datetime_begin"]);
+        }
+                
         try {
-            $query = $this->pdo->prepare("REPLACE INTO events (vCalendarFilename, ETag, datetime_begin, number_volunteers_required, vCalendarRaw) VALUES (:vCalendarFilename, :ETag, :datetime_begin, :number_volunteers_required, :vCalendarRaw)");
+            // can't use UPDATE INTO, because it would delete reminders related to the event (because of ON CASCADE DELETE)
```

Comment:
The comment is wrong, sorry. 
Should read: `can't use REPLACE INTO, because it would delete reminders related to the event (because of ON CASCADE DELETE)`

`REPLACE works exactly like INSERT, except that if an old row in the table has the same value as a new row for a PRIMARY KEY or a UNIQUE index, the old row is deleted before the new row is inserted. See Section 13.2.6, “INSERT Statement”` [MySQL doc](https://dev.mysql.com/doc/refman/8.0/en/replace.html).
Comment fixed in: 487f85f